Sources/SwiftBuildSupport/PackagePIFBuilder.swift

```diff
@@ -520,7 +522,15 @@ public final class PackagePIFBuilder {
         settings[.WATCHOS_DEPLOYMENT_TARGET] = builder.deploymentTargets[.watchOS] ?? nil
         settings[.DRIVERKIT_DEPLOYMENT_TARGET] = builder.deploymentTargets[.driverKit] ?? nil
         settings[.XROS_DEPLOYMENT_TARGET] = builder.deploymentTargets[.visionOS] ?? nil
-        settings[.DYLIB_INSTALL_NAME_BASE] = "@rpath"
+
+        for machoPlatform in [ProjectModel.BuildSettings.Platform.macOS, .macCatalyst, .iOS, .watchOS, .tvOS, .xrOS, .driverKit] {
+            settings.platformSpecificSettings[machoPlatform]![.DYLIB_INSTALL_NAME_BASE]! = ["@rpath"]
+        }
+        for elfPlatform in [ProjectModel.BuildSettings.Platform.linux, .android, .openbsd, .freebsd] {
+            // FIXME: Treating $ORIGIN as part of the soname is not really the right thing to do here
```

Comment:
Why not just avoid setting it at all, then? The soname on ELF platforms should just be the leaf name. $ORIGIN is only used in rpaths.

Reply:
Got this working but this patch now depends on https://github.com/swiftlang/swift-build/pull/604